runtime/lib/ttmetal/runtime.cpp

```
@@ -113,6 +126,39 @@ void dumpMemoryReport(Device deviceHandle) {
   }
 }
 
+std::unordered_map<tt::runtime::MemoryBufferType, tt::runtime::MemoryView>
+getMemoryView(Device deviceHandle, int deviceID) {
```

should we throw an error if the deviceID doesn't exist in the mesh?